### PR TITLE
Remove nonce check disabling

### PR DESCRIPTION
**Pull request by kstdl**
Need to be reviewed and merged first https://github.com/Taraxa-project/taraxa-evm/pull/102
#2082

### CODE REVIEW

File: tests/pbft_manager_test.cpp [hunk 1]
```diff
@@ -489,8 +494,9 @@ TEST_F(PbftManagerTest, check_get_eligible_vote_count) {
       if (nodes[i]->getDB()->getNumTransactionExecuted() != trxs_count) {
         std::cout << "node" << i << " executed " << nodes[i]->getDB()->getNumTransactionExecuted()
                   << " transactions. Expected " << trxs_count << std::endl;
-        auto dummy_trx = std::make_shared<Transaction>(nonce++, 0, gas_price, TEST_TX_GAS_LIMIT, bytes(),
-                                                       nodes[0]->getSecretKey(), nodes[0]->getAddress());
+        auto dummy_trx = make_transaction(0, nodes[0]->getAddress(), 0);
+        // std::make_shared<Transaction>(nonces[0]++, 0, gas_price, TEST_TX_GAS_LIMIT, bytes(),
+        //                                                nodes[0]->getSecretKey(), nodes[0]->getAddress());
```
Comment (JakubFornadel) [inline tests/pbft_manager_test.cpp:499]: remove this ?

File: tests/pbft_manager_test.cpp [hunk 2]
```diff
@@ -447,8 +453,9 @@ TEST_F(PbftManagerTest, check_get_eligible_vote_count) {
       if (nodes[i]->getDB()->getNumTransactionExecuted() != trxs_count) {
         std::cout << "node" << i << " executed " << nodes[i]->getDB()->getNumTransactionExecuted()
                   << " transactions, expected " << trxs_count << std::endl;
-        auto dummy_trx = std::make_shared<Transaction>(nonce++, 0, gas_price, TEST_TX_GAS_LIMIT, bytes(),
-                                                       nodes[0]->getSecretKey(), nodes[0]->getAddress());
+        auto dummy_trx = make_transaction(0, nodes[0]->getAddress(), 0);
+        // std::make_shared<Transaction>(nonces[0]++, 0, gas_price, TEST_TX_GAS_LIMIT, bytes(),
+        //                                                nodes[0]->getSecretKey(), nodes[0]->getAddress());
```
Comment (JakubFornadel) [inline tests/pbft_manager_test.cpp:458]: also this

File: tests/pbft_manager_test.cpp [hunk 3]
```diff
@@ -433,8 +437,10 @@ TEST_F(PbftManagerTest, check_get_eligible_vote_count) {
 
   const auto init_bal = node_1_expected_bal / nodes.size() / 2;
   for (size_t i(1); i < nodes.size(); ++i) {
-    auto master_boot_node_send_coins = std::make_shared<Transaction>(
-        nonce++, init_bal, gas_price, TEST_TX_GAS_LIMIT, bytes(), nodes[0]->getSecretKey(), nodes[i]->getAddress());
+    auto master_boot_node_send_coins = make_transaction(0, nodes[i]->getAddress(), init_bal);
+    // std::make_shared<Transaction>(
+    //     nonces[0]++, init_bal, gas_price, TEST_TX_GAS_LIMIT, bytes(), nodes[0]->getSecretKey(),
+    //     nodes[i]->getAddress());
```
Comment (JakubFornadel) [inline tests/pbft_manager_test.cpp:443]: remove commented code

File: tests/pbft_manager_test.cpp [hunk 4]
```diff
@@ -72,7 +72,7 @@ void check_2tPlus1_validVotingPlayers_activePlayers_threshold(size_t committee_s
   const auto init_bal = node_1_expected_bal / nodes.size();
   for (size_t i(1); i < nodes.size(); ++i) {
     auto master_boot_node_send_coins = std::make_shared<Transaction>(
-        nonce++, init_bal, gas_price, TEST_TX_GAS_LIMIT, bytes(), nodes[0]->getSecretKey(), nodes[i]->getAddress());
+        nonces[0]++, init_bal, gas_price, TEST_TX_GAS_LIMIT, bytes(), nodes[0]->getSecretKey(), nodes[i]->getAddress());
```
Comment (JakubFornadel) [inline tests/pbft_manager_test.cpp:75]: - should be here nonces[i]++ ? 
- same applies for multiple occurences of nonces[0]++ below

File: tests/final_chain_test.cpp [hunk 5]
```diff
@@ -308,53 +309,23 @@ TEST_F(FinalChainTest, nonce_test) {
...
+  auto trx6 = std::make_shared<Transaction>(6, 100, 0, 100000, dev::bytes(), sk, receiver_addr);
+  advance({trx4}, {false, false, true});
 
-  uint64_t skipped_nonce = 123;
-  cfg.chain.final_chain.state.execution_options.enable_nonce_skipping = true;
-  advance({std::make_shared<Transaction>(skipped_nonce, 100, 0, 100000, dev::bytes(), sk, receiver_addr)});
-  ASSERT_EQ(SUT->get_account(addr)->nonce.convert_to<uint64_t>(), skipped_nonce);
-}
-
-TEST_F(FinalChainTest, nonce_skipping) {
```
Comment (JakubFornadel) [inline tests/final_chain_test.cpp:345]: no more nonce skipping test ?

File: tests/final_chain_test.cpp [hunk 6]
```diff
@@ -308,53 +309,23 @@ TEST_F(FinalChainTest, nonce_test) {
...
   advance({trx1});
   advance({trx2});
   advance({trx3});
   advance({trx4});
 
-  // expecting transaction with nonce 2 after 1, so fail on 3
-  // execTransactionsOrder(sk, {{5, true}, {7, false}});
-}
-
-TEST_F(FinalChainTest, nonce_setting) {
```
Comment (JakubFornadel) [inline tests/final_chain_test.cpp:326]: no more nonce_setting test ?

File: libraries/core_libs/consensus/src/final_chain/state_api.cpp [hunk 7]
```diff
@@ -159,10 +159,10 @@ bytes StateAPI::get_code_by_address(EthBlockNumber blk_num, addr_t const& addr)
   return c_method_args_rlp<bytes, to_bytes, taraxa_evm_state_api_get_code_by_address>(this_c_, blk_num, addr);
 }
 
-ExecutionResult StateAPI::dry_run_transaction(EthBlockNumber blk_num, EVMBlock const& blk, EVMTransaction const& trx,
-                                              std::optional<ExecutionOptions> const& opts) const {
+ExecutionResult StateAPI::dry_run_transaction(EthBlockNumber blk_num, EVMBlock const& blk,
+                                              EVMTransaction const& trx) const {
```
Comment (JakubFornadel) [inline libraries/core_libs/consensus/src/final_chain/state_api.cpp:163]: const order

File: tests/pbft_manager_test.cpp (hunk 4 above)
Comment (kstdl) [reply tests/pbft_manager_test.cpp:75]: No. We should use nonce of sender(private key we specify). Maybe it is better to make some method\lambda for this like I did in other test

Comment (JakubFornadel) [reply tests/pbft_manager_test.cpp:75]: ah ok, [optional]: maybe just create something like const auto sender_idx = 0 and use that, which would resolve the confusion

**Comment by JakubFornadel** [comment]
I dont like this too, but lets fix const ordering for all of the functions that we somehow touch in our PRs...

**Review by JakubFornadel — APPROVED** [review]

File: tests/final_chain_test.cpp (hunk 6 above)
Comment (kstdl) [reply tests/final_chain_test.cpp:326]: I think `nonce_test` covers it

File: tests/final_chain_test.cpp (hunk 5 above)
Comment (JakubFornadel) [reply tests/final_chain_test.cpp:345]: what do you mean ? we must have nonce skipping enabled by default

**Review by mfrankovi — APPROVED** [review]

**Pull request merged**